pipeline/frontend/yaml/compiler/option.go

```diff
@@ -143,6 +143,17 @@ func WithProxy() Option {
 	)
 }
 
+// WithNetworks configures the compiler with additionnal networks
+// to be connected to build containers
+func WithNetworks(networks ...string) Option {
+	return func(compiler *Compiler) {
```

Comment:
also needs a simple unit test, similar to this https://github.com/cncd/pipeline/blob/master/pipeline/frontend/yaml/compiler/option_test.go#L38 😄